From 55f34d6a262a7e29172ad52ef2d6688ac2c02381 Mon Sep 17 00:00:00 2001
From: "Richard W.M. Jones" <rjones@redhat.com>
Date: Mon, 30 Dec 2013 20:32:03 +0000
Subject: [PATCH 11/11] Disable ocamldoc and camlp4opt (aarch64 only)

---
 Makefile | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/Makefile b/Makefile
index 10c80d2..e0a7d9e 100644
--- a/Makefile
+++ b/Makefile
@@ -686,8 +686,8 @@ alldepend::
 ocamldoc: ocamlc ocamlyacc ocamllex otherlibraries
 	cd ocamldoc && $(MAKE) all
 
-ocamldoc.opt: ocamlc.opt ocamlyacc ocamllex
-	cd ocamldoc && $(MAKE) opt.opt
+#ocamldoc.opt: ocamlc.opt ocamlyacc ocamllex
+#	cd ocamldoc && $(MAKE) opt.opt
 
 partialclean::
 	cd ocamldoc && $(MAKE) clean
@@ -734,8 +734,8 @@ alldepend::
 camlp4out: ocamlc ocamlbuild.byte
 	./build/camlp4-byte-only.sh
 
-camlp4opt: ocamlopt otherlibrariesopt ocamlbuild-mixed-boot ocamlbuild.native
-	./build/camlp4-native-only.sh
+#camlp4opt: ocamlopt otherlibrariesopt ocamlbuild-mixed-boot ocamlbuild.native
+#	./build/camlp4-native-only.sh
 
 # Ocamlbuild
 #ifeq ($(OCAMLBUILD_NOBOOT),"yes")
-- 
1.8.4.2

