--- ocaml-3.10.0/tools/Makefile.rpath	2007-06-02 16:53:10.000000000 +0200
+++ ocaml-3.10.0/tools/Makefile	2007-06-02 16:53:28.000000000 +0200
@@ -107,9 +107,6 @@
 	sed -e "s|%%BINDIR%%|$(BINDIR)|" \
             -e "s|%%SUPPORTS_SHARED_LIBRARIES%%|$(SUPPORTS_SHARED_LIBRARIES)|" \
             -e "s|%%MKSHAREDLIB%%|$(MKSHAREDLIB)|" \
-            -e "s|%%BYTECCRPATH%%|$(BYTECCRPATH)|" \
-            -e "s|%%NATIVECCRPATH%%|$(NATIVECCRPATH)|" \
-            -e "s|%%MKSHAREDLIBRPATH%%|$(MKSHAREDLIBRPATH)|" \
             -e "s|%%RANLIB%%|$(RANLIB)|" \
           ocamlmklib.mlp >> ocamlmklib.ml
 
