diff -up ocaml-3.10.0/otherlibs/labltk/support/cltkImg.c.tclver ocaml-3.10.0/otherlibs/labltk/support/cltkImg.c
--- ocaml-3.10.0/otherlibs/labltk/support/cltkImg.c.tclver	2008-01-04 18:56:47.000000000 +0100
+++ ocaml-3.10.0/otherlibs/labltk/support/cltkImg.c	2008-01-04 19:06:45.000000000 +0100
@@ -98,7 +98,7 @@ camltk_setimgdata_native (value imgname,
   pib.offset[0] = 0;
   pib.offset[1] = 1;
   pib.offset[2] = 2;
-  Tk_PhotoPutBlock(ph,&pib,Int_val(x),Int_val(y),Int_val(w),Int_val(h)
+  Tk_PhotoPutBlock(cltclinterp,ph,&pib,Int_val(x),Int_val(y),Int_val(w),Int_val(h)
 #if (TK_MAJOR_VERSION == 8 && TK_MINOR_VERSION >= 4 || TK_MAJOR_VERSION > 8)
                    , TK_PHOTO_COMPOSITE_SET
 #endif
diff -up ocaml-3.10.0/otherlibs/labltk/tkanim/tkAnimGIF.c.tclver ocaml-3.10.0/otherlibs/labltk/tkanim/tkAnimGIF.c
--- ocaml-3.10.0/otherlibs/labltk/tkanim/tkAnimGIF.c.tclver	2008-01-04 19:08:28.000000000 +0100
+++ ocaml-3.10.0/otherlibs/labltk/tkanim/tkAnimGIF.c	2008-01-04 19:08:37.000000000 +0100
@@ -334,7 +334,7 @@ FileReadGIF(interp, f, fileName, formatS
                 goto error;
             }
         }
-        Tk_PhotoPutBlock(photoHandle, &block, 0, 0, imageWidth, imageHeight
+        Tk_PhotoPutBlock(interp, photoHandle, &block, 0, 0, imageWidth, imageHeight
 #if (TK_MAJOR_VERSION == 8 && TK_MINOR_VERSION >= 4 || TK_MAJOR_VERSION > 8)
                    , TK_PHOTO_COMPOSITE_SET
 #endif
diff -up ocaml-3.10.0/configure.tclver ocaml-3.10.0/configure
--- ocaml-3.10.0/configure.tclver	2008-01-04 18:37:25.000000000 +0100
+++ ocaml-3.10.0/configure	2008-01-04 18:38:54.000000000 +0100
@@ -1342,6 +1342,7 @@ if test $has_tk = true; then
     8.2) tclmaj=8 tclmin=2 tkmaj=8 tkmin=2 ;;
     8.3) tclmaj=8 tclmin=3 tkmaj=8 tkmin=3 ;;
     8.4) tclmaj=8 tclmin=4 tkmaj=8 tkmin=4 ;;
+    8.5) tclmaj=8 tclmin=5 tkmaj=8 tkmin=5 ;;
     *) echo "This version is not known."; has_tk=false ;;
     esac
   else
